app/services/student_removal_service.rb

```diff
@@ -1,42 +1,63 @@
 # frozen_string_literal: true
 
 class StudentRemovalService
-  def initialize(students:, school:, remove_from_profile: false, token: nil)
-    @students = students
+  class NoSchoolError < StandardError; end
+  class NoClassesError < StandardError; end
+  class StudentHasProjectsError < StandardError; end
+  class NoopError < StandardError; end
```

Comment:
If you set up a `BaseError` class, then in the rake task you can just rescue anything
```ruby
  # Set up a BaseError class so we can rescue all errors
  BaseError = Class.new(StandardError)
  ## Specific errors
  NoSchoolError = Class.new(BaseError)
  NoClassesError = Class.new(BaseError)
  StudentHasProjectsError = Class.new(BaseError)
  NoopError = Class.new(BaseError)
```
then in the rake task

```
begin
  ...
rescue StudentRemovalService::BaseError => e
  # whatever it is here
end
```